tests/test_modeling_common.py

```diff
@@ -4027,6 +4027,101 @@ def test_flash_attn_2_from_config(self):
 
                 self.assertFalse(fa2_correctly_converted)
 
+    def _get_custom_4d_mask_test_data(self):
```

Comment:
This set of tests are now:
1. part of the mixing, so they are run on all push commits
2. a fast test, using `model = model_class(config)` from the test config
3. triggered by `model_class._supports_cache_class == True` -- recent LLMs [llama, cohere, gemma, mistral, mixtral, starcoder2, ...] have this attribute set to `True` and are 4D mask-compatible. Older models are often not compatible. Over time, as we spread the cache refactor, this test will be run on those classes as well 👀